### PR TITLE
Implement equals() and hashcode() on jakarta.mail.Header

### PR DESCRIPTION
**Pull request by jbescos**
https://github.com/eclipse-ee4j/mail/issues/594

### CODE REVIEW

**Comment by jbescos** [comment]
It fails because of copyright in unrelated files. Strangely, it does not fail when I run the same command locally:
`mvn -B -U -C -V clean verify org.glassfish.copyright:glassfish-copyright-maven-plugin:check -Poss-release,staging -Dgpg.skip=true `

**Comment by lukasj** [comment]
add `-Dcopyright.ignoreyear=true` to the build command in the GH action

File: api/src/main/java/jakarta/mail/Header.java [hunk 1]
```diff
@@ -67,4 +67,36 @@ public String getName() {
     public String getValue() {
 	return value;
     }
+
+    @Override
+    public int hashCode() {
+        final int prime = 31;
```
Comment (pzygielo) [inline api/src/main/java/jakarta/mail/Header.java:73]: Could `java.util.Objects.hash(name, value)` be used instead?

File: api/src/main/java/jakarta/mail/Header.java [hunk 2]
```diff
@@ -67,4 +67,36 @@ public String getName() {
...
+    @Override
+    public boolean equals(Object obj) {
+        if (this == obj)
+            return true;
+        if (obj == null)
+            return false;
+        if (getClass() != obj.getClass())
+            return false;
+        Header other = (Header) obj;
+        if (name == null) {
```
Comment (pzygielo) [inline api/src/main/java/jakarta/mail/Header.java:89]: Could `java.util.Objects.equals(name, other.name) && java.util.Objects.equals(value, other.value)` be used instead?

Comment (jbescos) [reply api/src/main/java/jakarta/mail/Header.java]: For the equals I have to verify first that `obj` is not null and it is the same class before using `Objects.equals`.

File: api/src/main/java/jakarta/mail/Header.java (hunk 1 above)
Comment (jbescos) [reply api/src/main/java/jakarta/mail/Header.java]: Sure

**Comment by jbescos** [comment]
Same error than here:

https://github.com/eclipse-ee4j/mail/pull/599#issuecomment-1087213759

**Review by lukasj — COMMENTED** [review]
put a note to changes, please

**Comment by jbescos** [comment]
> put a note to changes, please

Done

**Review by lukasj — APPROVED** [review]
LGTM

**Pull request merged**